app/services/puppet_fact_parser.rb

```diff
@@ -11,8 +11,8 @@ def operatingsystem
       args = {:name => os_name, :major => major, :minor => minor}
       os = Operatingsystem.find_or_initialize_by(args)
       if os_name[/debian|ubuntu/i] || os.family == 'Debian'
-        if facts[:lsbdistcodename]
-          os.release_name = facts[:lsbdistcodename]
+        if facts.dig(:os, :distro, :codename).presence || facts[:lsbdistcodename]
+          os.release_name = facts.dig(:os, :distro, :codename).presence || facts[:lsbdistcodename]
```

Comment:
Don't you need to actually use the value rather than the boolean?

```suggestion
          os.release_name = facts.dig(:os, :distro, :codename) || facts[:lsbdistcodename]
```

Reply:
"Returns the receiver if it's present otherwise returns nil. object.presence is equivalent to"

    object.present? ? object : nil

Reply:
In other words, empty string is ignored and eventually lead to an error. This pattern is used through the file, I thought it would be good idea to stick with it.

Reply:
dig already returns the object or nil, so it's redundant.

Reply:
Well, it's not redundant and other methods use it in this file. I suggest to stick with it, empty string is still irrelevant input and we want to skit that.

```
[1] pry(main)> h = {a: ""}
=> {:a=>""}
[2] pry(main)> h.dig(:a).presence
=> nil
[3] pry(main)> h.dig(:a)
=> ""
```